### PR TITLE
docs fix and v0.2.5

### PR DESCRIPTION
**Pull request by Trinkle23897**
Some documents need to be polished, so I made it WIP.

### CODE REVIEW

File: docs/index.rst [hunk 1]
```diff
@@ -28,25 +28,31 @@ Here is Tianshou's other features:
...
+
+    # latest release
+    pip install git+https://github.com/thu-ml/tianshou.git@master
+    # develop version
+    pip install git+https://github.com/thu-ml/tianshou.git@dev
 
 If you use Anaconda or Miniconda, you can install Tianshou through the following command lines:
-::
+
+.. code-block:: bash
```
Comment (youkaichao) [inline docs/index.rst]: Maybe adding a ``$`` for bash command is better, just like ">>>" in python.

File: tianshou/utils/net/common.py [hunk 2]
```diff
@@ -44,17 +45,22 @@ class Recurrent(nn.Module):
...
-        self.fc1 = nn.Linear(np.prod(state_shape), 128)
-        self.fc2 = nn.Linear(128, np.prod(action_shape))
+        self.fc1 = nn.Linear(np.prod(state_shape), size)
+        self.fc2 = nn.Linear(size, np.prod(action_shape))
 
     def forward(self, s, state=None, info={}):
+        """In the evaluation mode, s should be with shape ``[bsz, dim]``; in
+        the training mode, s should be with shape ``[bsz, len, dim]``. See the
+        code and comment for more detail.
+        """
```
Comment (youkaichao) [inline tianshou/utils/net/common.py]: ``size`` is over-used. ``hidden_units`` may be better. The same in the other files.

Comment (youkaichao) [reply tianshou/utils/net/common.py]: just a reference:

```python
from sklearn.neural_network import MLPClassifier
MLPClassifier?
Init signature:
MLPClassifier(
    hidden_layer_sizes=(100,),
...
```

So, maybe ``hidden_layer_size`` is better.

File: docs/test_docs.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    if os.path.exists('_build'):
+        shutil.rmtree('_build')
+    p = Popen(['make', 'html'], stderr=PIPE)
+    _, err = p.communicate()
+    err = [i for i in err.decode().splitlines() if 'WARNING' in i]
+    assert len(err) == 0, '\n'.join(err)
+
+
+if __name__ == '__main__':
+    main()
```
Comment (youkaichao) [inline docs/test_docs.py]: maybe sphinx has some built-in functions to check this. This file is too brute-force.

Comment (Trinkle23897) [reply docs/test_docs.py]: I think so, but the sphinx makefile only provides "make doctest" which is not what our expected.

Comment (youkaichao) [reply docs/test_docs.py]: Simply remove the ``sphinx.ext.imgmath`` extension in conf.py works just fine.

**Review by youkaichao — APPROVED** [review]

File: README.md [hunk 4]
```diff
@@ -49,24 +49,27 @@ In Chinese, Tianshou means divinely ordained and is derived to the gift of being
 Tianshou is currently hosted on [PyPI](https://pypi.org/project/tianshou/). It requires Python >= 3.6. You can simply install Tianshou with the following command:
 
 ```bash
-pip3 install tianshou
+$ pip install tianshou
```
Comment (duburcqa) [inline README.md:52]: using `pip` is not recommended. Use `python -m pip` instead.

File: README.md [hunk 5]
```diff
@@ -49,24 +49,27 @@ In Chinese, Tianshou means divinely ordained and is derived to the gift of being
...
 ```
 
 You can also install with the newest version through GitHub:
 
 ```bash
-pip3 install git+https://github.com/thu-ml/tianshou.git@master
+# latest release
+$ pip install git+https://github.com/thu-ml/tianshou.git@master
+# develop version
+$ pip install git+https://github.com/thu-ml/tianshou.git@dev
```
Comment (duburcqa) [inline README.md:61]: Same

File: README.md [hunk 6]
```diff
@@ -49,24 +49,27 @@ In Chinese, Tianshou means divinely ordained and is derived to the gift of being
...
 ```bash
-pip3 install tianshou
+$ pip install tianshou
 ```
 
 You can also install with the newest version through GitHub:
 
 ```bash
-pip3 install git+https://github.com/thu-ml/tianshou.git@master
+# latest release
```
Comment (duburcqa) [inline README.md:58]: Or, "bleeding edge version". As you like.

File: README.md [hunk 7]
```diff
@@ -82,9 +85,9 @@ If no error occurs, you have successfully installed Tianshou.
 
 The tutorials and API documentation are hosted on [tianshou.readthedocs.io](https://tianshou.readthedocs.io/).
 
-The example scripts are under [test/](https://github.com/thu-ml/tianshou/blob/master/test) folder and [examples/](https://github.com/thu-ml/tianshou/blob/master/examples) folder. It may fail to run with PyPI installation, so please re-install the github version through `pip3 install git+https://github.com/thu-ml/tianshou.git@master`.
+The example scripts are under [test/](https://github.com/thu-ml/tianshou/blob/master/test) folder and [examples/](https://github.com/thu-ml/tianshou/blob/master/examples) folder.
 
-中文文档位于 [https://tianshou.readthedocs.io/zh/latest/](https://tianshou.readthedocs.io/zh/latest/)
+中文文档位于 [https://tianshou.readthedocs.io/zh/latest/](https://tianshou.readthedocs.io/zh/latest/)。
```
Comment (duburcqa) [inline README.md:90]: What is this weird dot at the end of line? x)

File: README.md [hunk 8]
```diff
@@ -182,17 +185,12 @@ Define some hyper-parameters:
...
-gamma = 0.9
-n_step = 3
-eps_train, eps_test = 0.1, 0.05
-epoch = 10
-step_per_epoch = 1000
-collect_per_step = 10
-target_freq = 320
-batch_size = 64
+lr, epoch, batch_size = 1e-3, 10, 64
 train_num, test_num = 8, 100
```
Comment (duburcqa) [inline README.md:189]: I don't like assignment dispatch, but as you want.

File: README.md [hunk 9]
```diff
@@ -254,26 +252,26 @@ collector.close()
...
 
 ## Citing Tianshou
 
 If you find Tianshou useful, please cite it in your publications.
 
 ```latex
 @misc{tianshou,
-  author = {Jiayi Weng, Minghao Zhang, Dong Yan, Hang Su, Jun Zhu},
+  author = {Jiayi Weng, Minghao Zhang, Alexis Duburcq, Kaichao You, Dong Yan, Hang Su, Jun Zhu},
   title = {Tianshou},
```
Comment (duburcqa) [inline README.md:275]: Your way to cite Tianshou is for publication, not software. It is not very appropriate.

File: README.md (hunk 7 above)
Comment (youkaichao) [reply README.md:90]: It is the period in Chinese. 😄

File: README.md (hunk 4 above)
Comment (youkaichao) [reply README.md:52]: Why is that ``pip`` is not recommended?

File: docs/tutorials/tictactoe.rst [hunk 10]
```diff
@@ -162,8 +162,8 @@ Tianshou already provides some builtin classes for multi-agent learning. You can
 
 Random agents perform badly. In the above game, although agent 2 wins finally, it is clear that a smart agent 1 would place an ``x`` at row 4 col 4 to win directly. 
 
-Train a MARL Agent
-------------------
+Train an MARL Agent
```
Comment (duburcqa) [inline docs/tutorials/tictactoe.rst:165]: Why "an"?

File: README.md (hunk 7 above)
Comment (duburcqa) [reply README.md:90]: Nice :)

File: README.md (hunk 4 above)
Comment (duburcqa) [reply README.md:52]: I don't know. It is a good question. You could probably found the answer on stackoverflow.com :)

File: docs/tutorials/tictactoe.rst (hunk 10 above)
Comment (youkaichao) [reply docs/tutorials/tictactoe.rst:165]: @Trinkle23897 reads it as M-A-R-L, so he uses "an". I read it as MA-R-L, so I use "a".

File: README.md (hunk 9 above)
Comment (Trinkle23897) [reply README.md:275]: Yes, we are submitting tianshou to JMLR and we would like to place your name into the authorlist.

File: README.md (hunk 4 above)
Comment (Trinkle23897) [reply README.md:52]: I agree the way to use python -m, but it may cause extra confusion for new people I think.

File: README.md (hunk 9 above)
Comment (duburcqa) [reply README.md:275]: That would be awesome :smiley:

File: README.md (hunk 4 above)
Comment (duburcqa) [reply README.md:52]: I have no problem with `pip` either. It was just to make sure you are aware of the new syntax.

**Review by duburcqa — APPROVED** [review]

**Pull request merged**